### PR TITLE
connect messages resource to frontend

### PR DESCRIPTION
**Pull request by yndajas**
## Dev Summary
[Mocks](https://www.figma.com/file/fYkreARKdHsMh8iPSh7dti/GC-prototype-v1?node-id=0%3A453)

Proof of concept for connecting messages backend resource to the messages components on the frontend:
- temporarily turned off JWT auth guards for a few endpoints/routes to make it easier to test for now
- in the `MessagesInboxView` component, updated the method to fetch messages, which returns an array of objects with the keys matching the Message type. *TODO:* fetch by transaction

## Test Plan

preReqs:

[ ] following command succeeds `cd client && npm run checks && cd ../server && npm run checks`

repro steps:

#### create two users and note IDs

```
curl --request POST \
 --url http://localhost:3001/api/users \
 --header 'Content-Type: application/json' \
 --data '{
   "first_name": "Smino",
   "last_name": "Grigio",
   "email": "christopher@smi.no",
   "password": "nohoopti"
   }'
```

```
 curl --request POST \
  --url http://localhost:3001/api/users \
  --header 'Content-Type: application/json' \
  --data '{
    "first_name": "Noname",
    "last_name": "Goatlife",
    "email": "fatimah@no.name",
      "password": "bookclubfactorybaby"
    }'
```

Each request should return something in the format of: `{"first_name":"TestFirst","last_name":"TestLast","email":"test@test.com","id":1}`

#### create messages

paste the ID from the last step in each request

```
curl --request POST \
--url http://localhost:3001/api/messages \
-H 'Content-Type: application/json' \
--data '{ "text": "hello", "transaction_id": "1", "user": {"id": PASTE_FIRST_USER_ID_HERE} }'
```

```
curl --request POST \
--url http://localhost:3001/api/messages \
-H 'Content-Type: application/json' \
--data '{ "text": "hi", "transaction_id": "1", "user": {"id": PASTE_SECOND_USER_ID_HERE} }'
```

```
curl --request POST \
--url http://localhost:3001/api/messages \
-H 'Content-Type: application/json' \
--data '{ "text": "I'\''m interested in helping out your NP", "transaction_id": "1", "user": {"id": PASTE_FIRST_USER_ID_HERE} }'
```

Each request should return something in the format of:
`{"text":"hello","transaction_id":"1","user":{"id":2},"id":1,"created_date":"2021-12-19T20:17:42.343Z"}`

### in client

- load up client
- log in to client as second user
- go to messages

It should look something like this:

Screenshot 2021-12-14 at 19 58 17

### CODE REVIEW

**Comment by jd2rogers2** [comment]
> [Mocks](link): Figma is only letting me click around - there doesn't seem to be any node linking option for me

try clicking around on to a bunch of different objects in the mocks, you should see the `node-id` url search param change
that's the inner page anchor (or whatever it's called)

File: client/src/MessageInboxView.tsx [hunk 1]
```diff
@@ -99,14 +100,32 @@ const fetchTransactions = (): Promise<Transaction[]> => {
...
+            text: message.text,
+            transactionId: message.transaction_id,
+            user: {
+                id: message.user_id,
+                firstName: message.user_first_name
+            }
+        }
+    })
+    
+    return await messages;
```
Comment (jd2rogers2) [inline client/src/MessageInboxView.tsx:120]: await only needed on asynchronous lines
shouldn't need it on reduce or here

**Review by jd2rogers2 — COMMENTED** [review]
hopefully this helps get you unblocked
ping me if you have any questions on it or if it doesn't work. if either we can set up a call

**Comment by yndajas** [comment]
> > [Mocks](link): Figma is only letting me click around - there doesn't seem to be any node linking option for me
> 
> try clicking around on to a bunch of different objects in the mocks, you should see the `node-id` url search param change that's the inner page anchor (or whatever it's called)

@jd2rogers2 ah, the problem was you need to create and sign into a Figma account in order for it to start giving you the node ID in the URL - might be worth adding that clarification to the guidance?

Will take a look another look at the PR itself soon to see if I can get it in a working proof of concept state using all messages

File: server/src/messages/dto/create-message.dto.ts [hunk 2]
```diff
@@ -3,7 +3,12 @@ import { User } from 'src/users/entities/user.entity';
 import { Message } from '../entities/message.entity';
 
 export class CreateMessageDto extends PartialType(Message) {
+  // TODO: use real user and pass this through to the client
+  // making sure the MessageInboxView component can still fetch messages properly
+
```
Comment (jd2rogers2) [inline server/src/messages/dto/create-message.dto.ts:8]: does this comment still apply? i think we are already doing this with the below `user: User` property

File: server/src/messages/dto/create-message.dto.ts [hunk 3]
```diff
@@ -3,7 +3,12 @@ import { User } from 'src/users/entities/user.entity';
...
   text: string;
   user: User;
+  // user_id: number;
+  // user_first_name: string;
```
Comment (jd2rogers2) [inline server/src/messages/dto/create-message.dto.ts:12]: don't think we need these anymore

File: server/src/messages/messages.controller.ts [hunk 4]
```diff
@@ -10,14 +10,17 @@ import { DeleteResult } from 'typeorm';
...
-    console.log(createMessageDto);
     return this.messagesService.create(createMessageDto);
   }
 
-  @UseGuards(JwtAuthGuard)
+  // TODO: reinstate UseGuards here and in the Delete action when
+  // ready - currently these are needed to add data to test the
+  // MessageInboxView component
+
+  // @UseGuards(JwtAuthGuard)
```
Comment (jd2rogers2) [inline server/src/messages/messages.controller.ts:23]: actually, remove these altogether
i'm working on something that'll replace these anyways
(will use cookie guard in the future https://github.com/Nonprofit-Exchange-Hub/web-app/pull/91/files)

File: server/src/users/users.controller.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ export class UsersController {
     return user;
   }
 
-  @UseGuards(JwtAuthGuard)
+  // @UseGuards(JwtAuthGuard)
```
Comment (jd2rogers2) [inline server/src/users/users.controller.ts:21]: let's remove this endpoint altogether as well
make sure nothing hits it first but i don't see a need for it right now

File: client/src/MessageInboxView.tsx [hunk 6]
```diff
@@ -192,7 +195,7 @@ function MessageInboxView(): JSX.Element {
                 Re: {selectedTransaction.asset.title}
               </Typography>
               {messages?.map((m) => (
-                <MessageCard message={m} isCurrentUser={m.user.id === user?.id} />
+                <MessageCard message={m} isCurrentUser={m.user.id === user?.user.id} />
```
Comment (jd2rogers2) [inline client/src/MessageInboxView.tsx:198]: so i know this was my bad but this `user.user` is ugly
let's undo this change and make the change in auth.service to be
`return this.jwtService.sign(user, { secret: jwtConstants.secret });`
and that SHOULD still work. lmk if not

**Review by jd2rogers2 — APPROVED** [review]
some nits to address first but go ahead and 🚢  once you've addressed
nice work!! 
next step adding a post/fetch from here to the submit button

**Pull request merged**